core/src/helpers.rs

```diff
@@ -260,7 +260,14 @@ pub fn hash_to_field_element(hash: H256) -> Result<FieldElement, KakarotClientEr
 }
 
 pub fn bytes_to_felt_vec(bytes: Bytes) -> Vec<FieldElement> {
-    bytes.to_vec().into_iter().map(FieldElement::from).collect()
+    let bytes_vec = bytes.to_vec();
```

Comment:
why are we doing this refactor?
here, our goal is doing 1 byte = 1 felt 
Bytes -> Vec<u8> -> Vec<FieldElement>

Reply:
Yes, but here, since we are dealing with bytes, we know the value will be between 0 and 255 (since bytes.to_vec() will yield a Vec<u8>)